### PR TITLE
CASMPET-5640 Corrected VCS Password Update instructions (release/1.0)

### PR DESCRIPTION
**Pull request by atifsyedali**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

Corrected VCS password update instructions. No code changes.

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

VCS password update instructions have stayed the same since 1.0 to mainline dev so not expecting incompatibilities.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMPET-5640](https://jira-pro.its.hpecorp.net:8443/browse/CASMPET-5640)
* Change will also be needed in `release/1.2` and `main`
* ~Future work required by [issue id](issue link)~
* ~Documentation changes required in [issue id](issue link)~
* ~Merge with/before/after `<insert PR URL here>`~

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `fanta`

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

1. Followed instructions to change VCS password and then verified the change by logging into VCS.
2. Change the VCS password back.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? No
- Were continuous integration tests run? If not, why?  No
- Was upgrade tested? If not, why?  No
- Was downgrade tested? If not, why?  No
- Were new tests (or test issues/Jiras) created for this change?  No

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

No other known issues.

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 1]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.
+
+   The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`
+
+   Use the following admin login credentials:
+
+   - Username: `admin`
+   - The password can be obtained with the following command:
+
+    ```bash
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:45]: Indent this code block under the above bullet

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 2]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+   Point the browser at `https://vcs.SHASTA_CLUSTER_DNS_NAME/vcs/user/settings/account`.
+
+   If presented with Keycloak login, use `crayvcs` as the username and the new VCS password. Wait to be redirected to the Gitea login page before continuing to the next step.
+
+1. Use the following Gitea login credentials:
+
+   - Username: `crayvcs`
+   - The old VCS password, which can be obtained with the following command:
+
+    ```bash
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:73]: Indent this code block under the above bullet

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:73]: Indent this code block under the above bullet

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 3]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
-```bash
-ncn# kubectl create secret generic vcs-user-credentials --save-config \
---from-literal=vcs_username="crayvcs" \
---from-literal=vcs_password="NEW_PASSWORD" \
---dry-run=client -o yaml | kubectl apply -f -
-```
-The `NEW_PASSWORD` value must be replaced with the updated password.
+1. Log in to Keycloak with the default admin credentials.
+
+   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:36]:
```suggestion
   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing `SYSTEM_DOMAIN_NAME` with the actual NCN's DNS name.
```
```suggestion
   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:36]:
```suggestion
   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing `SYSTEM_DOMAIN_NAME` with the actual NCN's DNS name.
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 4]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+
+   The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:38]:
```suggestion
   The following is an example URL for a system: `https://auth.cmn.system1.us.cray.com/keycloak/admin`
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:38]:
```suggestion
   The following is an example URL for a system: `https://auth.cmn.system1.us.cray.com/keycloak/admin`
```
```suggestion
   The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 5]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+
+1. In the `Search...` textbox, type in `crayvcs` and click the search icon.
+
+1. In the filtered table below, click on the ID for the row that shows `crayvcs` in the `Username` column. 
+
+1. Go to the `Credentials` tab and change the password.
+
+   Enter the new password in the `Reset Password` form. Ensure `Temporary` is switched **off**. Click on `Reset Password` button.
+
+1. Now log in to Gitea with the default admin credentials.
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:62]:
```suggestion
1. Log in to Gitea with the default `admin` credentials.
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:62]:
```suggestion
1. Log in to Gitea with the default `admin` credentials.
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 6]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
---dry-run=client -o yaml | kubectl apply -f -
-```
-The `NEW_PASSWORD` value must be replaced with the updated password.
+1. Log in to Keycloak with the default admin credentials.
+
+   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.
+
+   The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`
+
+   Use the following admin login credentials:
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:40]:
```suggestion
   Use the following `admin` login credentials:
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:40]:
```suggestion
   Use the following `admin` login credentials:
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 7]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+
+   - Username: `crayvcs`
+   - The old VCS password, which can be obtained with the following command:
+
+    ```bash
+    ncn# kubectl get secret -n services vcs-user-credentials \
+            --template={{.data.vcs_password}} | base64 --decode
+    ```
+
+1. Enter the existing password (from previous step), new password and confirmation, and then click `Update Password`.
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:78]:
```suggestion
1. Enter the existing password (from previous step), new password, and confirmation, and then click `Update Password`.
```
```suggestion
1. Enter the existing password (from previous step), new password and confirmation, and then click `Update Password`.
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:78]:
```suggestion
1. Enter the existing password (from previous step), new password, and confirmation, and then click `Update Password`.
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 8]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+1. Now SSH into `ncn-w001` or `ncn-m001`.
+1. Run `git clone https://github.com/Cray-HPE/csm.git`.
+1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to your desired working directory, and run the following commands from that working directory.
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:81]:
```suggestion
1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to the desired working directory.
1. Change directories to be in the working directory set in the previous step.
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:81]:
```suggestion
1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to the desired working directory.
1. Change directories to be in the working directory set in the previous step.
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 9]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+   ncn# kubectl create secret -n loftsman generic site-init --from-file=customizations.yaml
+   ```
+
+1. Encrypt the values after changing the `customizations.yaml` file.
+
+    ```bash
+    ncn# ./secrets-seed-customizations.sh customizations.yaml
+    ```
+
+   If the above command complains that it cannot find `certs/sealed_secrets.crt` then you can run the following commands to create it:
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:124]:
```suggestion
   If the above command complains that it cannot find `certs/sealed_secrets.crt`, then you can run the following commands to create it:
```
```suggestion
   If the above command complains that it cannot find `certs/sealed_secrets.crt` then you can run the following commands to create it:
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:124]:
```suggestion
   If the above command complains that it cannot find `certs/sealed_secrets.crt`, then you can run the following commands to create it:
```

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 10]
```diff
@@ -27,15 +29,166 @@ The initial VCS login credentials for the `crayvcs` user are stored in three pla
...
+To change the password in the `vcs-user-credentials` Kubernetes secret, use the following procedure:
 
-```bash
-ncn# kubectl create secret generic vcs-user-credentials --save-config \
---from-literal=vcs_username="crayvcs" \
---from-literal=vcs_password="NEW_PASSWORD" \
---dry-run=client -o yaml | kubectl apply -f -
-```
-The `NEW_PASSWORD` value must be replaced with the updated password.
+1. Log in to Keycloak with the default admin credentials.
```
Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:34]:
```suggestion
1. Log in to Keycloak with the default `admin` credentials.
```

Comment (nrockershousen) [inline operations/configuration_management/Version_Control_Service_VCS.md:34]:
```suggestion
1. Log in to Keycloak with the default `admin` credentials.
```

**Review by nrockershousen — CHANGES_REQUESTED** [review]

**Comment by atifsyedali** [comment]
@nrockershousen Thanks for your review and suggestions!

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Same changes as this PR: https://github.com/Cray-HPE/docs-csm/pull/1661

**Comment by atifsyedali** [comment]
> Same changes as this PR: #1661

Done

**Review by mharding-hpe — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

**Comment by atifsyedali** [comment]
I don't seem to have permissions to merge. Can someone please squash and merge this?

File: operations/configuration_management/Version_Control_Service_VCS.md [hunk 11]
```diff
@@ -4,52 +4,218 @@ The Version Control Service \(VCS\) includes a web interface for repository mana
...
+    ```
+
+1. Edit the `sysmgmt.yaml` to only include the `gitea` chart and all its current data.
+
+   Example:
+
+    ```yaml
+    apiVersion: manifests/v1beta1
+      metadata:
+        name: sysmgmt
```
[the remark this answers is not in the document]
Comment (atifsyedali) [reply operations/configuration_management/Version_Control_Service_VCS.md:166]: @kburns-hpe Made changes as you suggested. Can you please review? Thanks

[the remark this answers is not in the document]
Comment (kburns-hpe) [reply operations/configuration_management/Version_Control_Service_VCS.md:171]: looks good.

**Review by kburns-hpe — APPROVED** [review]

**Pull request merged**